### PR TITLE
Added support for nnet3 raw computation without acoustic model

### PR DESCRIPTION
**Pull request by vimalmanohar**
1) Created nnet3-get-egs-dense-targets for continuous valued targets
2) Created NnetSimpleComputer class that does the nnet computation part of DecodableAmNnetSimple and used in nnet3-compute. DecodableAmNnetSimple is made a child class of NnetSimpleComputer.

### CODE REVIEW

File: src/nnet3bin/nnet3-compute.cc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  // device, but it probably won't make a huge difference in speed for typical
+  // setups.
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+
+    const char *usage =
+        "Propagate through neural network model\n"
+        "Usage: nnet3-compute [options] <nnet-in> <features-rspecifier> <matrix-wspecifier>\n";
```
Comment (danpovey) [inline src/nnet3bin/nnet3-compute.cc]: should say <raw-nnet-in>

Comment (vimalmanohar) [inline src/nnet3bin/nnet3-compute.cc]: I didn't understand. What is the change I should make?

Comment (danpovey) [inline src/nnet3bin/nnet3-compute.cc]: <nnet-in> to <raw-nnet-in>

File: src/nnet3/nnet-am-decodable-simple.cc [hunk 2]
```diff
@@ -31,20 +33,12 @@ DecodableAmNnetSimple::DecodableAmNnetSimple(
     const VectorBase<BaseFloat> *ivector,
     const MatrixBase<BaseFloat> *online_ivectors,
     int32 online_ivector_period):
+    NnetSimpleComputer(opts.simple_computer_opts, am_nnet.GetNnet(), feats, am_nnet.LeftContext(), am_nnet.RightContext(), ivector, online_ivectors, online_ivector_period),
```
Comment (danpovey) [inline src/nnet3/nnet-am-decodable-simple.cc]: line too long

File: src/nnet3/nnet-am-decodable-simple.cc [hunk 3]
```diff
@@ -54,18 +48,12 @@ DecodableAmNnetSimple::DecodableAmNnetSimple(
     const MatrixBase<BaseFloat> &feats,
     const MatrixBase<BaseFloat> &ivectors,
     int32 online_ivector_period):
+    NnetSimpleComputer(opts.simple_computer_opts, am_nnet.GetNnet(), feats, am_nnet.LeftContext(), am_nnet.RightContext(), NULL, &ivectors, online_ivector_period),
```
Comment (danpovey) [inline src/nnet3/nnet-am-decodable-simple.cc]: line too long

File: src/nnet3/nnet-am-decodable-simple.cc [hunk 4]
```diff
@@ -74,21 +62,14 @@ DecodableAmNnetSimple::DecodableAmNnetSimple(
     const AmNnetSimple &am_nnet,
     const MatrixBase<BaseFloat> &feats,
     const VectorBase<BaseFloat> &ivector):
+    NnetSimpleComputer(opts.simple_computer_opts, am_nnet.GetNnet(), feats, am_nnet.LeftContext(), am_nnet.RightContext(), &ivector, NULL, 0),
```
Comment (danpovey) [inline src/nnet3/nnet-am-decodable-simple.cc]: line too long

**Comment by vimalmanohar** [comment]
I have made the changes.

**Comment by danpovey** [comment]
Thanks!

**Pull request merged**